KsApi/ServiceType.swift

```diff
@@ -255,13 +255,17 @@ public protocol ServiceType {
   func markAsRead(messageThread: MessageThread) -> SignalProducer<MessageThread, ErrorEnvelope>
 
   /// Posts a comment to a project.
-  func postComment(_ body: String, toProject project: Project)
+  func deprecatedPostComment(_ body: String, toProject project: Project)
     -> SignalProducer<DeprecatedComment, ErrorEnvelope>
 
   /// Posts a comment to an update.
-  func postComment(_ body: String, toUpdate update: Update)
+  func deprecatedPostComment(_ body: String, toUpdate update: Update)
     -> SignalProducer<DeprecatedComment, ErrorEnvelope>
 
+  /// Posts a comment to a project or replies in a thread
+  func postComment(input: PostCommentInput)
+    -> SignalProducer<GraphMutationPostCommentResponseEnvelope, GraphError>
```

Comment:
So I actually think we should return `Comment` here and do the data transformation in KsApi. In the same way that we've hidden the deserialization of the comments pagination from the app.

Reply:
Oh I've also moved away from `GraphError` in other places and just returned `ErrorEnvelope`. The idea is also that once we implement Apollo all of this implementation detail is hidden and the `ServiceType` interface won't have to change in the places that we've already done this.